pkg/manager/impl/validation/shared_execution.go

```diff
@@ -0,0 +1,34 @@
+package validation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/flyteorg/flyteadmin/pkg/common"
+	"github.com/flyteorg/flyteadmin/pkg/errors"
+	"github.com/flyteorg/flyteadmin/pkg/repositories"
+	repoInterfaces "github.com/flyteorg/flyteadmin/pkg/repositories/interfaces"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flytestdlib/logger"
+)
+
+func ValidateCluster(ctx context.Context, db repositories.RepositoryInterface, executionID *core.WorkflowExecutionIdentifier, cluster string) error {
```

Comment:
May be a doc that this validate executionId belongs to passed in cluster string.

Also can we mention somewhere why its ok to have workflowExecution.Cluster different than common.DefaultProducerID and still raise no error

Reply:
Already done here: https://github.com/flyteorg/flyteadmin/pull/307/files#diff-89fdbc388b31a2407358d353ba0dc95fb7b9a4862b0528fc0792aabb67e8b178R15 but I can copy the comment

Reply:
Ok . that should be good then.